packages/playwright/src/reporter.ts

```diff
@@ -78,34 +78,35 @@ export interface ReplayPlaywrightConfig
 
 class ReplayPlaywrightReporter implements Reporter {
   reporter?: ReplayReporter<ReplayPlaywrightRecordingMetadata>;
-  captureTestFile = ["1", "true"].includes(
-    process.env.PLAYWRIGHT_REPLAY_CAPTURE_TEST_FILE?.toLowerCase() || "true"
-  );
+  captureTestFile: boolean;
+  config: ReplayPlaywrightConfig;
+
+  constructor(config: ReplayPlaywrightConfig) {
+    if (!config || typeof config !== "object") {
+      throw new Error(
+        `Expected an object for @replayio/playwright/reporter configuration but received: ${config}`
+      );
+    }
 
-  getTestId(test: TestCase) {
-    return test.titlePath().join("-");
+    this.config = {
+      ...config,
+      apiKey: config.apiKey || process.env.REPLAY_API_KEY || process.env.RECORD_REPLAY_API_KEY,
+    };
+    if (!this.config.apiKey) {
+      throw new Error(
+        "`@replayio/playwright/reporter` requires an API key. Either pass a value to the apiKey plugin configuration or set the REPLAY_API_KEY environment variable"
```

Comment:
`config.apiKey` is only required if `config.upload` is `true`.

Reply:
That crossed my mind but then I noticed that we also need `apiKey` for various graphql calls. I see now that they are guarded and we don't perform them when there is no API key.

How does it work when those calls are not performed? What the user might be missing in such a case? When it comes to the metadata, I see that it's stored in our `recordings.log` - so at least that is not completely lost.

Reply:
I don't know how much sense it makes to run the tests with our reporter but without an API key. But since we've been offering that option until now I guess we should keep supporting it.